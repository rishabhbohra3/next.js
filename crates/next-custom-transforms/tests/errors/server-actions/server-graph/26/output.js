/* __next_internal_action_entry_do_not_use__ {"803128060c414d59f8552e4788b846c0d2b7f74743":"$$RSC_SERVER_CACHE_0"} */ import { registerServerReference } from "private-next-rsc-server-reference";
import { encryptActionBoundArgs, decryptActionBoundArgs } from "private-next-rsc-action-encryption";
import { cache as $$cache__ } from "private-next-rsc-cache-wrapper";
import { BaseClass } from './base';
export var $$RSC_SERVER_CACHE_0 = $$cache__("default", "803128060c414d59f8552e4788b846c0d2b7f74743", 0, /*#__TURBOPACK_DISABLE_EXPORT_MERGING__*/ async function bar() {
    // arguments is not allowed here
    console.log(arguments);
    // this is not allowed here
    return this.foo();
});
Object.defineProperty($$RSC_SERVER_CACHE_0, "name", {
    "value": "bar",
    "writable": false
});
export class MyClass extends BaseClass {
    static async foo() {
        // super is allowed here
        super.foo();
        return fetch('https://example.com').then((res)=>res.json());
    }
    static bar = registerServerReference($$RSC_SERVER_CACHE_0, "803128060c414d59f8552e4788b846c0d2b7f74743", null);
}
