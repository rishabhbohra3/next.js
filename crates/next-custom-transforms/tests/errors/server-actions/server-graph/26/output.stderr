  x "use cache" functions can not use `super`.
  | 
    ,-[input.js:14:1]
 13 |     // super is not allowed here
 14 |     super.bar()
    :     ^^^^^
 15 |     // arguments is not allowed here
    `----
  x "use cache" functions can not use `arguments`.
  | 
    ,-[input.js:16:1]
 15 |     // arguments is not allowed here
 16 |     console.log(arguments)
    :                 ^^^^^^^^^
 17 |     // this is not allowed here
    `----
  x "use cache" functions can not use `this`.
  | 
    ,-[input.js:18:1]
 17 |     // this is not allowed here
 18 |     return this.foo()
    :            ^^^^
 19 |   }
    `----
